e2e/regression/regression_test.go

```diff
@@ -79,11 +92,24 @@ func TestRegression(t *testing.T) {
 			require.NoError(err)
 			require.NoError(os.WriteFile(path.Join(ct.WorkDir, "resources.yml"), resourceBytes, 0o644))
 
-			deploymentName, _ := strings.CutSuffix(file.Name(), ".yml")
+			resourceName, _ := strings.CutSuffix(file.Name(), ".yml")
+
+			unstructuredResources, err := kubeapi.UnmarshalUnstructuredK8SResource(readYaml)
+			require.NoError(err)
 
 			t.Cleanup(func() {
-				// delete the deployment
-				require.NoError(ct.Kubeclient.Client.AppsV1().Deployments(ct.Namespace).Delete(context.Background(), deploymentName, metav1.DeleteOptions{})) //nolint:usetesting // see https://github.com/ldez/usetesting/issues/4
+				bgDeletion := metav1.DeletePropagationBackground
+				for _, r := range unstructuredResources {
+					client, err := ct.Kubeclient.ResourceInterfaceFor(r)
+					if err != nil {
+						t.Log("error creating client for resource deletion: ", err)
+					}
+					if err := client.Delete(context.WithoutCancel(t.Context()), resourceName, metav1.DeleteOptions{
```

Comment:
I think we should be using a real context with a short timeout here. `WithoutCancel` makes this effectively `context.Background`.